# Quantitative and computational biology

GSK Graduate School, Spring 2017

### Location: 

Zuckerman Research Center (**ZRC-138**)

### Summary: 

Students in this course will learn to apply quantitative exploratory
data analysis techniques to different forms of experimental data. The
course will begin with a software skills “bootcamp”, which will
introduce students to computing via the UNIX shell, and to computing in
the R programming language. The remaining lessons will be a blend of
practical skills and theoretical concepts. Students will become
comfortable performing exploratory data analysis, and will understand
how concepts from statistics underlie the tools they use. These skills
will enable them both to collaborate effectively with computational
biologists, as well as to carry out their own computational experiments.

Specific topics include: practical aspects of data formatting and
management; visualization of data and model parameters; an introduction
to probability, elementary statistics and hypothesis testing;
experimental design and tools for differential expression analysis in
RNA-seq and ChIP-seq; interpreting model fit, correct interpretation of
statistical evidence, and multiple hypothesis testing correction
methods; common data structures for working with biological sequences;
enrichment testing for ranked gene sets; common bioinformatics tools and
data quality assessment tools; introduction to structural biology and
tools for visualization.

Towards the end of the course, students are encouraged to do a project
on bioinformatic data analysis and present their results in the last
class.

### Course Instructors:
- Sonya Hanson ([*sonya.hanson@choderalab.org*](mailto:sonya.hanson@choderalab.org))
- Hatice Ulku Osmanbeyoglu ([*uosmanbey@gmail.com*](mailto:uosmanbey@gmail.com))
- Yuri Pritykin ([*pritykin@cbio.mskcc.org*](mailto:pritykin@cbio.mskcc.org))
- Lee Zamparo ([*zamparol@cbio.mskcc.org*](mailto:zamparol@cbio.mskcc.org))

### Teaching Fellow:
- Robert Bowman ([*bowmanr@mskcc.org*](mailto:bowmanr@mskcc.org))

### Assessment:

This class will not be graded. However, periodic student surveys in
class and online are planned to assess if the presented material is
properly learnt. All students will also be asked to complete a survey at
the end of the course, to obtain feedback on the course and assess if
this experimental format is successful and what changes need to be done
when the course is offered in future years.

### Lecture dates (February 4, 2017 – Monday, May 22, 2017)

One bootcamp: 
- Saturday, February 4, 2017, 9:00 AM – 5:00 PM

Fifteen class sessions:
- Thirteen sessions on Mondays 1:00 PM – 3:00 PM (later on 3-5pm)
   - (February 6, February 13, February 27, March 6, March 13, March 20, April 3, April 10, April 24, May 1, May 8, May 15 and May 22)
- Two sessions on Fridays from 1:00 PM – 3:00 PM 
   - (March 31 and April 28)

### Bootcamp details:

 **(Saturday, February 4, 2017)** 

   Basic tools for research computing. Installation of shell, git, R and  
   Rstudio. Introduction to shell computing, the R programming language,  
   basic statistics and visualization.

### Class Sessions:

1.  **Lab 1** (Monday, February 6, 2017)

   Review of loading and plotting data in R. 
   
2. **Lab 2** (Monday, February 13, 2017)

   Introduction to statistics and probability. 

3.  **Lab 3** (Monday, February 27, 2017)

   Review of basic stats, mean and standard deviation, normal distribution, other types of distributions.
   
4. **Lab 4** (Monday, March 6, 2017)

   Introduction to hypothesis testing, review of plotting techniques.
   
5. **Lab 5** (Monday, March 13, 2017)

   Genomic data structures in R and Bioconductor: GenomicRanges, GenomicFeatures, Biostrings, BSgenome, AnnotationDbi.

6. **Lab 6** (Monday, March 20, 2017)

   Practical RNA-seq with DESeq.
   
7. **Lab 7** (Friday, March 31, 2017)

   Practical ChIP-seq, peak calling with MACS2, viewing output data in genome browsers.   

8. **Lab 8** (Monday, April 3, 2017)

   Efficent ways to source data from public repositories, discussion of project ideas.

9.  **Lab 9** (Monday, April 10, 2017)

   Interpreting P-values and Multiple Hypothesis Testing.

10. **Lab 10** (Monday, April 24, 2017

   Bioinformatics tools: browsers, GO enrichment, gene set enrichment analysis. 

11. **Lab 11** (Friday, April 28, 2017)

   Exploratory data analysis by PCA, clustering, t-SNE.

12. **Lab 12** (Monday, May 1, 2017)

   Sequence Alignment and Chemoinformatics/PDB tools

13. **Lab 13** (Monday, May 8, 2017)

   In class consultation about projects.

14. **Lab 14** (Monday, May 15, 2017)

   In class consultation about projects.
   
15. **Lab 15** (Monday, May 22, 2017)
   
   Project presentations.

---

### Books and Materials: 

Students will need a laptop computer on which they can install software
(R and R Studio), and bring to class. All software used in the course is
free, and run on recent versions of Linux, Mac OS X, and Microsoft
Windows.

R: [*http://www.r-project.org*](http://www.r-project.org)

Rstudio: [*http://www.rstudio.com*](http://www.rstudio.com/)

Gitshell (Windows only):
[*https://git-for-windows.github.io/*](https://git-for-windows.github.io/)

The course does not require the use of a specific textbook, but instead
makes use of the following online resources:

**Software Capentry**

-   [*Introduction to the Shell*](http://swcarpentry.github.io/shell-novice/)

-   [*R for Reproducible Scientific Analysis*](http://swcarpentry.github.io/r-novice-gapminder/)

**qBio-I course materials (courtesy of Jason Banfelder)**

-   [*http://physiology.med.cornell.edu/people/banfelder/qbio/schedule\_2015/\#qbio-i*](http://physiology.med.cornell.edu/people/banfelder/qbio/schedule_2015/#qbio-i)

**Biomedical Data Science (courtesy of Raphael Irizarry and Michael
Love)**

-   [*http://genomicsclass.github.io/book/*](http://genomicsclass.github.io/book/)


